espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -201,39 +198,15 @@ def forward(self, xs_pad, ilens, ys_pad):
             hs_len = hs_mask.view(batch_size, -1).sum(1)
             loss_ctc = self.ctc(hs_pad.view(batch_size, -1, self.adim), hs_len, ys_pad)
 
-            cers, char_ref_lens = [], []
-
-            y_hats = self.ctc.argmax(hs_pad.view(batch_size, -1, self.adim)).data
-            for i, y in enumerate(y_hats):
-                y_hat = [x[0] for x in groupby(y)]
-                y_true = ys_pad[i]
-
-                seq_hat = [self.char_list[int(idx)] for idx in y_hat if int(idx) != -1]
-                seq_true = [self.char_list[int(idx)] for idx in y_true if int(idx) != -1]
-                seq_hat_text = "".join(seq_hat).replace(self.space, ' ')
-                seq_hat_text = seq_hat_text.replace(self.blank, '')
-                seq_true_text = "".join(seq_true).replace(self.space, ' ')
-
-                hyp_chars = seq_hat_text.replace(' ', '')
-                ref_chars = seq_true_text.replace(' ', '')
-                if len(ref_chars) > 0:
-                    cers.append(editdistance.eval(hyp_chars, ref_chars))
-                    char_ref_lens.append(len(ref_chars))
-
-            cer_ctc = float(sum(cers)) / sum(char_ref_lens) if cers else None
+            ys_hat = self.ctc.argmax(hs_pad.view(batch_size, -1, self.adim)).data
+            cer_ctc = self.error_calculator(ys_hat, ys_pad, is_ctc=True)
 
         # 5. compute cer/wer
-        if self.training or not (self.report_cer or self.report_wer):
+        if self.training or self.error_calculator is None:
             cer, wer = 0.0, 0.0
```

Comment:
This may confuse people.
`None` is better?
At least `0.0` is wrong.

Reply:
Ok.
I was planning to apply it but in the other PR. #774. Should I apply this here? or keep it as part of #774.

Reply:
Oh, that's better. Let's do it there.